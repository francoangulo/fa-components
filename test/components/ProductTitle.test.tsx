import * as React from "react";

import { ProductTitle } from "../../src/components/";
import renderer from "react-test-renderer";
import { ProductCard } from "../../src/components/ProductCard";
import { product1 } from "../data/products";

describe("ProductTitle", () => {
  test("Renders ProcutTitle with a custom title without crashing", () => {
    const wrapper = renderer.create(
      <ProductTitle title="Custom title" className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe("ProductTitle", () => {
  test("Renders ProcutTitle with a the title of the product without crashing", () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
