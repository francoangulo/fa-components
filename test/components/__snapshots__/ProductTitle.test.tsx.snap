// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Renders ProcutTitle with a custom title without crashing 1`] = `
<span
  className="productDescription custom-class"
>
  Custom title
</span>
`;

exports[`ProductTitle Renders ProcutTitle with a the title of the product without crashing 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card No Image
  </span>
</div>
`;
